components/ciphertool/src/main/java/org/wso2/ciphertool/cipher/AsymmetricCipher.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.ciphertool.cipher;
+
+import org.apache.commons.lang.StringUtils;
+import org.wso2.ciphertool.exception.CipherToolException;
+import org.wso2.ciphertool.utils.Constants;
+import org.wso2.ciphertool.utils.KeyStoreUtil;
+import org.wso2.ciphertool.utils.Utils;
+
+import java.nio.charset.StandardCharsets;
+import java.security.InvalidKeyException;
+import java.security.Key;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.UnrecoverableKeyException;
+import java.security.cert.Certificate;
+import java.util.Base64;
+
+import javax.crypto.Cipher;
+import javax.crypto.NoSuchPaddingException;
+
+/**
+ * Provides methods for encryption and decryption using asymmetric key algorithms.
+ */
+public class AsymmetricCipher implements CipherMode {
+
+    private final String keyAlias;
+    private final KeyStore keyStore;
+    private final Cipher cipher;
+    private static final String CIPHER_INIT_ERROR_MESSAGE = "Error initializing Cipher.";
```

Comment:
Normally these error messages should be come from a separate error message class.